tests/conftest.py

```diff
@@ -91,3 +93,11 @@ def cte_optimization_enabled(pytestconfig):
 
 def pytest_sessionstart(session):
     os.environ["SNOWPARK_LOCAL_TESTING_INTERNAL_TELEMETRY"] = "1"
+
+
+@pytest.fixture(autouse=True)
+def clear_warning_dict():
+    yield
+    # clear the warning dict so that warnings from one test don't affect
```

Comment:
Our tests are run in parallel using pytest-xdist, I don't think this logic ensures a clean warning_dict for each test?

Reply:
Thanks for bringing this up. pytest-xdist runs tests in separate processes that run in parallel, so there's only one `warning_dict` in each process. This fixture resets the warning dict before every test case that each process runs, so `warning_dict` should be clear before each test case.